lib/purposes/ControllerProofPurpose.js

```diff
@@ -32,6 +32,22 @@ module.exports = class ControllerProofPurpose extends ProofPurpose {
     }
   }
 
+  /**
+   * Validates the purpose of a proof. This method is called during
+   * proof verification, after the proof value has been checked against the
+   * given verification method (e.g. in the case of a digital signature, the
+   * signature has been cryptographically verified against the public key).
+   *
+   * @param proof
```

Comment:
seems like maybe a @typedef or something is in order so there is no need to repeat what a `documentLoader` or `expansionMap` are 3 or 4 times.

Reply:
Ah yeah, I need to look into those..

Reply:
@mattcollier I'd like to deal with `@typedef` in another PR, if at all possible.